Add warning about tool scripts needing editor restart (#10544)
godotengine/godot-docs: tutorials/plugins/running_code_in_the_editor.rst

Save the script and return to the editor. You should now see your object rotate.
If you run the game, it will also rotate.

> .. danger::

Author: Thanks :)

>     You may need to restart the editor. This is a known bug found in all Godot 4 versions.

Author: ```suggestion
    You may need to restart the editor. This is a known bug that has been resolved with Godot 4.4.
```
As there currently is an open MR to resolve this, we could alternatively use this phrasing as some users might use an older version of godot.

Author: ```suggestion
    You may need to restart the editor. This is a known bug that has not been resolved by Godot 4.3.
```
second alternative phrasing, not as positive, but maybe better until we know for sure?

Author: I dunno a lot of people use older versions of softwares that do not auto-update or are hesitant to update things like game engines because they are afraid the games they are working on might break.

But I agree with you, if a maintainer didn't suggest it, I would not have created this.
